### PR TITLE
[FlyDSL][fix] (tuned_gemm): detach GEMM inputs before FlyDSL DLPack export

### PR DESCRIPTION
**Pull request by kangwangamd**
## Summary

flydsl 0.3.2 (#4912) exports kernel args via tensor.__dlpack__(), which raises BufferError("Can't export tensors that require gradient") on grad-tracking tensors. 
gpt-oss CUDA-graph capture on gfx950 hits this and crashes the server. 
Detach inp/weights/bias before flydsl_hgemm.

## Validation

Verified on MI355X: candidate 636098e5 + flydsl 0.3.2 crashes unpatched, serves clean with this patch.

### CODE REVIEW

**Comment by xytpai** [comment]
Please move it to: https://github.com/ROCm/aiter/blob/main/aiter/ops/flydsl/kernels/gemm_a16w16_gfx950.py#L1392

```python
def _dynamic_tensor_arg(tensor, leading_dim):
    if getattr(tensor, "requires_grad", False):
        tensor = tensor.detach()
    return flyc.from_dlpack(tensor).mark_layout_dynamic(leading_dim=leading_dim)
```

Here -> https://github.com/ROCm/aiter/pull/5327

**Comment by xytpai** [comment]
Move to https://github.com/ROCm/aiter/pull/5327
